docs/operating-procedures.md

```diff
@@ -50,43 +50,49 @@ The number of `idr-omeroreadonly*` servers may vary, check `/etc/hosts` oni `idr
-script will return a list of all the IDs for the first Image in each Fileset:
+Bio-Formats cache files should be regenerated for each imported fileset. For large databases, this operation can be parallelized across the available OMERO
+servers (read-write and read-only). 
 
-    $ echo "copy (select r.i from (select f.id, i.id as i, rank()
-                  over (partition by f.id order by i.id) as x
-                  from fileset f, image i where f.id = i.fileset)
-                  as r where r.x = 1) to stdout;" > first_image_per_fileset.sql
-
-For large databases, cache regeneration should be run in a parallelized manner
-across the available OMERO servers (read-write and read-only):
+Using the OMERO read-write server as the master node, the following commands
+create a list of all remote servers, creates a local SSH key and copies it 
+across nodes to prevent `PermissionsDenied` errors:
 
     $ grep -oE 'omero[^ ]+$' /etc/hosts > nodes
-
-To prevent `PermissionDenied`, create a SSH key and copy it across nodes:
-
     $ ssh-keygen
     $ for i in $(cat nodes); do ssh-copy-id -i ~/.ssh/id_rsa $i; done
 
 Bio-Formats cache regeneration makes use of the
-[OMERO.CLI render plugin](https://pypi.org/project/omero-cli-render/). First
-of all, make sure the plugin is installed on each OMERO server:
+[OMERO.CLI render plugin](https://pypi.org/project/omero-cli-render/). The plugin can be installed on each node as:
 
     $ for i in $(cat nodes); do ssh $i 'pip install --user omero-cli-render'; done
 
 The existing Bio-Formats cache can be moved from the read-write server as:
 
-    $ ssh omeroreadwrite sudo mkdir /data/BioFormatsCache/backup
-    $ ssh omeroreadwrite sudo mv /data/BioFormatsCache/* /data/BioFormatsCache/backup
+    $ sudo mkdir /data/BioFormatsCache/backup
+    $ sudo mv /data/BioFormatsCache/* /data/BioFormatsCache/backup
+
+For very large databases with millions of filesets, it is preferable to
+distribute the cache regeneration at the level of plates and datasets. The
+following SQL will generate a file `ids.txt` containing a list of all the
+objects:
+
+    $ bin/omero hql --limit -1 --ids-only --style csv 'select MIN(field.image.id) FROM WellSample AS field GROUP BY field.well.plate' > plates.txt
+    $ bin/omero hql --limit -1 --ids-only --style csv 'select d.id from Dataset d' > datasets_ids.txt
+    $ cat first_ids_per_plate.txt | cut -d ',' -f2 | sed -e 's/^/Image:/' > ids.txt 
+    $ cat datasets_ids.txt | cut -d ',' -f2 | sed -e 's/^/Dataset:/' >> ids.txt
+
+The cache file regeneration can be started using the GNU parallel utility 
+using the following command:
 
-Then execute the cache file regeneration using GNU parallel from the proxy
-servers across all nodes using the `ids.txt` list generated by the SQL script:
+    $ parallel --eta --sshloginfile nodes -a ids.txt --results /tmp/cache/ -j10 '/opt/omero/server/OMERO.server/bin/omero render -s localhost -u public -w public test --force'
```

Comment:
start command in screen since it will take few days
e.g.
```
screen -dmS cache parallel --eta --sshloginfile nodes -a ids.txt --results /tmp/cache/ -j10 '/opt/omero/server/OMERO.server/bin/omero render -s localhost -u public -w public test --force'
```

Reply:
See 6f18610